### PR TITLE
docs(contributing): add supporting context for new contributors

### PR DESCRIPTION
**Pull request by geospatialem**
**Related Issue:** n/a

## Summary
- Refactors section on how contributors can help, as a few items have surfaced during Hacktoberfest 🎃👻🦇 contributions 
  - Issues
  - Dev work
  - Resources 
- Adds more context on why contributors should comment on issues (so we can assign issues to them)
- Updates the jsbin url to reflect the latest versions (linked the same in our bug issue template)
- Small copyedits
- Adds conventional commit example and link to our recent commits for additional examples for folks to reference

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -56,9 +56,10 @@ An issue can only have one of the lifecycle labels at any time. Please make sure
 
 ### Issues that cannot be worked on
 
-There are three labels that mean an issue is not ready for development:
+There are four labels that mean an issue is not ready for development:
 
 - `design`: Issues that need design consultation, such as interaction research/feedback, visual mockups, and general approval. Once design completes their review, an additional label, `ready for dev` will be added to the issue, which means a developer can pick up the issue.
+- `spike`: Issues that need to research a question or resolve a complex task with uncertain outcomes. Once the spike has been performed a `spike complete` label is added to the issue, which means a developer can pick up the issue.
```
Comment (jcfranco) [inline CONTRIBUTING.md:62]: Should we add more info behind the spike workflow or is this best suited elsewhere?

**Review by jcfranco — APPROVED** [review]
👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻
👻🎃👻👻👻🎃👻👻🎃🎃👻👻🎃🎃🎃👻👻🎃🎃🎃👻🎃🎃🎃🎃👻🎃👻
👻🎃🎃👻👻🎃👻🎃👻👻🎃👻👻🎃👻👻🎃👻👻👻👻🎃👻👻👻👻🎃👻
👻🎃👻🎃👻🎃👻🎃👻👻🎃👻👻🎃👻👻🎃👻👻👻👻🎃🎃🎃👻👻🎃👻
👻🎃👻👻🎃🎃👻🎃👻👻🎃👻👻🎃👻👻🎃👻👻👻👻🎃👻👻👻👻👻👻
👻🎃👻👻👻🎃👻👻🎃🎃👻👻🎃🎃🎃👻👻🎃🎃🎃👻🎃🎃🎃🎃👻🎃👻
👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻👻

File: CONTRIBUTING.md (hunk 1 above)
Comment (geospatialem) [reply CONTRIBUTING.md:62]: @jcfranco We don't currently list specifics on either the ready for dev or spike workflows outside of the scripts. I'm not sure if we should add it outside of the workflows, but do you think it'd be good to share more widely?

Comment (jcfranco) [reply CONTRIBUTING.md:62]: Gotcha. In retrospect, I think it's better to keep it high level here.

**Pull request merged**